server/lib/validation/languages/tok.js

```diff
@@ -0,0 +1,70 @@
+const tokenizeWords = require('talisman/tokenizers/words');
+
+const TRANSLATION_KEY_PREFIX = 'TRANSLATION_KEY:';
+
+// Minimum of words that qualify as a sentence.
+const MIN_WORDS = 1;
+
+// Maximum of words allowed per sentence to keep recordings in a manageable duration.
+const MAX_WORDS = 18;
+
+const INVALIDATIONS = [{
+  fn: (sentence) => {
+    const words = tokenizeWords(sentence);
+    return words.length < MIN_WORDS || words.length > MAX_WORDS;
+  },
+  error: `${TRANSLATION_KEY_PREFIX}sc-validation-number-of-words`,
+}, {
+  regex: /[0-9]+/,
+  error: `${TRANSLATION_KEY_PREFIX}sc-validation-no-numbers`,
+}, {
+  regex: /[<>+*#@%^[\]()/]/,
+  error: `${TRANSLATION_KEY_PREFIX}sc-validation-no-symbols`,
+}, {
+  // capital letters at start of word only
+  regex: /[\w\.][A-Z]/,
+  error: 'sitelen suli li ken lon open nimi taso (ike: "jan AwiPota"; pona: "jan Awipota", "jan Awi Pota")',
+},
+
+// The following invalidations are to make all submissions conform to Toki Pona's
+// phonotactics. Any words that don't follow the phonotactics would have ambiguous
+// pronunciations, and some speakers might struggle pronouncing them.
+
+{
+  // No non-Toki-Pona letters
+  regex: /[BbCcDdFfGgHhQqRrVvXxYyZzĈĜĤĴŜŬĉĝĥĵŝŭäÄöÖüÜßððÀÁÂÃÅÆÇÈÉÊËÌÍİÎÏÐÑÒÓÔÕØÙÚÛÛÝŽàáâãåæçèéêëìíîïðñòóôõøùúûýþÿāăąćċčďđēĕėęěğġģħĩīĭįıķĸĺļľŀłńņņṫšЎḃḋḟṁṗṡẁẃẅẛỳαβΓγΔδεζηΘθικΛλμνΞξΠπρΣσςτșếōůūŁşşǐżőňựňžịŌŏČŠřś]/,
```

Comment:
I think it would make sense to negate this and only list valid letters. I think that's what the suggestion from @tbodt is, but I don't understand Toki Pona, so can't say for sure :)

Reply:
Indeed. I couldn't figure out a good regex to allow punctuation as well, though.

Reply:
Yeah, we would probably need to specify an exhaustive list of all valid characters, including punctuation, for that to work.

The currently list is copied and slightly modified from Esperanto's validation script, which is meant to only allow Esperanto letters. After tbodt's comment I had the idea that it might be better to specify entire blocks of codepoints for this. E.g. `[BbCcDdFfGgHhQqRrVvXxYyZz\u00C0-\u02BF\u1E00-\u1EFF]` That should weed out the vast majority of Latin script letter variants.

And actually, we might also want to add `\uF1900-\uF19FF`, the codepoints to which Toki Pona's sitelen pona script is assigned [in the UCSUR standard](http://www.kreativekorp.com/ucsur/charts/sitelen.html).

Reply:
I changed it to `[BbCcDdFfGgHhQqRrVvXxYyZz\u00C0-\u02BF\u1E00-\u1EFF\uF1900-\uF19FF]` for now, when I updated the translations part

Reply:
I made a similar change to Esperanto's script, changing `[wWqQxXyYäÄöÖüÜßððÀÁÂÃÅÆÇÈÉÊËÌÍİÎÏÐÑÒÓÔÕØÙÚÛÛÝŽàáâãåæçèéêëìíîïðñòóôõøùúûýþÿāăąćċčďđēĕėęěğġģħĩīĭįıķĸĺļľŀłńņņṫšЎḃḋḟṁṗṡẁẃẅẛỳαβΓγΔδεζηΘθικΛλμνΞξΠπρΣσςτșếōůūŁşşǐżőňựňžịŌŏČŠřś]` to a more clean and more complete `[qQwWxXyYÀ-ćĊ-ěĞ-ģĞ-ģĦ-ĳĶ-śŞ-ūŮ-\u02AF\u1E00-\u1EFFα-ωΑ-ΩЀ-ӿ]`.
And I also imported Esperanto translations from Pontoon while I was at it.